CONTRIBUTING.md

```diff
@@ -15,42 +14,103 @@
 
-Since using TestContainers to launch containers shouldn't be bundled into browser under test, starting the WildFly container is being done using Cypress arbitrary Node code running handle - [`cy.task`](https://docs.cypress.io/api/commands/task), use 
 ```
 cy.startWildFlyContainer
 ```
+
 method, which is a wrapper for
+
 ```
 cy.task("start:wildfly:container")
 ```
-to start the container. This method/task also automatically sets the deployed HAL container endpoint in the `allowed-origins` attribute. Both of the approaches would return the management endpoint of the deployed WildFly instance.
+
+to start the container. Note that despite starting the web server being marked as an anti-pattern, starting the WildFly server clean has its benefits in terms of a clean environment per test and also it introduces means for test parallelization (we know what we're doing :slightly_smiling_face:). This method/task also automatically sets the deployed HAL container endpoint in the `allowed-origins` attribute. Both of the approaches would return the management endpoint of the deployed WildFly instance.
 
 ### Communicating with WildFly management interface
 
 Communication with the WildFly management interface is also being done using Cypress task, `execute:cli` to be precise. A better example of running `/system-property=to-add:add(value=myVal)` in JBoss CLI in Berg using `cy.task`
-would be 
+would be
+
 ```
 cy.task("execute:cli", {
       managementApi: `${managementEndpoint}/management`,
       operation: "add",
       address: ["system-property", "to-add"],
       value: "myVal"
     });
-```
\ No newline at end of file
+```
+
+### Test template
+
+To start developing tests, you can copy paste following example into your `packages/testsuite/cypress/e2e/test-to-be-added.cy.ts`
+
+```typescript
+// describe method is an equivalent of a TestClass
+describe("TESTS: Path => To => Tested => Resource", () => {
+  const configurationFormId = "resource-form";
+
+  // This is and before method are very important
+  let managementEndpoint: string;
+
+  // Here we start our WildFly container prior to test execution
+  before(() => {
+    // Obtains WildFly management endpoint, that will be used
+    // for backend data retrieval and validation
+    // cy.startWildflyContainer also starts the WildFly container under
+    // the name of the spec/test file, e.g in our case: test_to_be_added
+    cy.startWildflyContainer().then((result) => {
+      managementEndpoint = result as string;
+    });
+  });
+
+  // Here we stop & remove started WildFly container(s)
+  after(() => {
+    cy.task("stop:containers");
+  });
+
+  // it methods is an equivalent of a @Test annotated method,
+  // First argument is the name of the test, second is a lambda test function
+  it("Edit default-extended-persistence-inheritance", () => {
+    // Navigates to the UI page
+    // http://localhost:<mapped_hal_port>?connect=http://localhost:<mapped_wildfly_port>#<token>
```

Comment:
I think it's quite understandable what the commands do and such comments are not much helpful. As a developer I would rather appreciate where I can find more such methods so that I see the full pallet and possible arguments/parameters.